File: AuctionSystem.Data/Model/Auction.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace AuctionSystem.Data.Model
{
    public partial class Auction
    {
        public Auction()
        {
            Bids = new HashSet<Bid>();
            Products = new HashSet<Product>();
        }

        public Guid Id { get; set; }
        public string Title { get; set; }
        public Guid UserId { get; set; }
        public DateTime PublishedOn { get; set; }
        public string Description { get; set; }
        public int? StateId { get; set; }

        public virtual AuctionState State { get; set; }
        public virtual User User { get; set; }
        public virtual ICollection<Bid> Bids { get; set; }
        public virtual ICollection<Product> Products { get; set; }
    }
}

File: AuctionSystem.Client/src/models/auction.js
export default class Auction {
    constructor(id, title, description, publishedOn, leadingBid, endDate, stateId, minBid) {
        this.id = id
        this.title = title
        this.description = description
        this.publishedOn = publishedOn
        this.leadingBid = leadingBid
        this.endDate = endDate
        this.stateId = stateId
        this.minBid = minBid
    }
}